### PR TITLE
⚖️ Fix available staking balances

### PR DESCRIPTION
**Pull request by thesan**
#3319

~It looks like non-rivalrous locked funds are not included in the transferable balance (`DeriveBalancesAll[DeriveBalancesAll]` in polkadotjs):~
The same fund can be staked under multiple locks at the same time:
Screenshot from 2022-07-28 15-44-03

~Although voting is the only non-rivalrous lock where the stake needs to be set manually, I also fixed the validation on announce  candidacy, and apply for role.~

➕ improved `useBalance` a bit because it was making the API unavailable for some length of time. That said I thought that we had taken care of that already... did we revert this change before for some reason ?

### CODE REVIEW

File: packages/ui/src/accounts/components/SelectAccount/OptionAccount.tsx [hunk 1]
```diff
@@ -11,21 +11,24 @@ import { Colors } from '@/common/constants'
...
   return (
     <>
       <AccountInfo account={option} locked={isLocked} />
       <BalanceInfoInRow>
-        <InfoTitle>Transferable balance</InfoTitle>
+        <InfoTitle>{balanceType} balance</InfoTitle>
         <InfoValueWithLocks>
-          <Value value={balance?.transferable} locked={isLocked} />
-          <AccountLocks locks={balance?.locks} />
+          <Value value={balance ?? balances?.total} locked={isLocked} />
```
Comment (thesan) [inline packages/ui/src/accounts/components/SelectAccount/OptionAccount.tsx:30]: Is this on prupose ?

```suggestion
          <Value value={balance} locked={isLocked} />
```

File: packages/ui/src/council/modals/AnnounceCandidacy/helpers.ts [hunk 2]
```diff
@@ -29,8 +29,8 @@ export interface AnnounceCandidacyFrom {
 export const baseSchema = Yup.object().shape({
   staking: Yup.object().shape({
     account: StakingAccountSchema.required('This field is required'),
-    amount: BNSchema.test(minContext('Minimal stake amount is ${min} tJOY', 'minStake'))
-      .test(maxContext('Insufficient funds to cover staking', 'controllerAccountBalance'))
+    amount: BNSchema.test(maxContext('Insufficient funds to cover staking', 'balances.transferable'))
```
Comment (thesan) [inline packages/ui/src/council/modals/AnnounceCandidacy/helpers.ts:32]:
```suggestion
    amount: BNSchema.test(maxContext('Insufficient funds to cover staking', 'balances.total'))
```

File: packages/ui/src/working-groups/modals/ApplyForRoleModal/helpers.ts [hunk 3]
```diff
@@ -21,7 +21,10 @@ export const baseSchema = Yup.object().shape({
     account: StakingAccountSchema.required(),
     roleAccount: AccountSchema.required(),
     rewardAccount: AccountSchema.required(),
-    amount: Yup.number().test(minContext('You need at least ${min} stake', 'minStake')).required('Amount is required'),
+    amount: Yup.number()
+      .test(minContext('You need at least ${min} stake', 'minStake'))
+      .test(maxContext('Insufficient funds to cover staking', 'balances.transferable'))
```
Comment (thesan) [inline packages/ui/src/working-groups/modals/ApplyForRoleModal/helpers.ts:26]:
```suggestion
      .test(maxContext('Insufficient funds to cover staking', 'balances.total'))
```

**Review by WRadoslaw — APPROVED** [review]
LGTM

**Pull request merged**